File: Assignments/Assignment 37/C Codes/program37_1.c
// 1.Write a program which accept string from user and accept one character. 
// Check whether that character is present in string or not.

// Input : "Marvellous Multi OS"
//         e

// Output: true

// Input:  "Marvellous Multi OS"
//         W

// Output: false

#include<stdio.h>
#include<stdbool.h>

bool ChkChar(char *str, char ch)
{
    bool bAns = false;

    while(*str != '\0')
    {
        if(*str == ch)
        {
            return bAns = true;
            break;
        }
        
        str++;
    }

    return bAns;
}

int main()
{
    char Arr[50] = {"\0"};
    char cValue = '\0';
    bool bRet = false;

    printf("Enter String : \n");
    scanf("%[^'\n']", Arr);

    fflush(stdin);

    printf("Enter the character : \n");
    scanf("%c", &cValue);

    bRet = ChkChar(Arr, cValue);

    if(bRet == true)
    {
        printf("Character found\n");
    }
    else
    {
        printf("Character not found\n");
    }

    return 0;
}

File: Assignments/Assignment 37/Java Codes/program37_1.java
// 1.Write a program which accept string from user and accept one character. 
// Check whether that character is present in string or not.

// Input : "Marvellous Multi OS"
//         e

// Output: true

// Input:  "Marvellous Multi OS"
//         W

// Output: false

import java.util.Scanner;

class Test
{
    boolean ChkChar(String str, char ch)
    {
        int iCnt = 0;
        boolean bAns = false;

        char Arr[] = str.toCharArray();

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            if(Arr[iCnt] == ch)
            {
                bAns = true;
                break;
            }
        }

        return bAns;
    }
}

class program37_1
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        boolean bRet = false;

        System.out.println("Enter String : ");
        String Arr = sobj.nextLine();

        System.out.println("Enter character : ");
        char cValue = sobj.nextLine().charAt(0);

        Test tobj = new Test();

        bRet = tobj.ChkChar(Arr, cValue);

        if(bRet == true)
        {
            System.out.println("Character found");
        }
        else
        {
            System.out.println("Character not found");
        }
    }
}